### PR TITLE
Allow compression of chunks with aborted inserts

### LINKED ISSUE #492: Can't compress chunks with aborted inserts
Because of this, we cannot compress chunks if an insert failed:

<pre>
bool ChunkCompressionTask::chunk_is_completed(const Chunk& chunk, const uint32_t max_chunk_size) {
  if (chunk.size() != max_chunk_size) return false;

  auto mvcc_columns = chunk.mvcc_columns();
<strong>
  for (const auto begin_cid : mvcc_columns->begin_cids) {
    if (begin_cid == Chunk::MAX_COMMIT_ID) return false;
  }
</strong>
  return true;
}
</pre>

I added `ChunkCompressionTaskTest.CompressionWithAbortedInsert` to document this.

A possible (!) solution would be for a rollback of an insert to set the begin and end cids to 0 (instead of infty). This would document that the rows are completed, but will not make them visible to anyone. Since MVCC is a tricky thing this should not be implemented without a more careful discussion. @mjendruk - do you have any thoughts on this?

### PR DESCRIPTION
**Pull request by mrks**
fixes #492

### CODE REVIEW

File: src/lib/concurrency/transaction_manager.cpp [hunk 1]
```diff
@@ -16,12 +16,12 @@ TransactionManager& TransactionManager::get() {
 void TransactionManager::reset() {
   auto& manager = get();
   manager._next_transaction_id = 1u;
-  manager._last_commit_id = 0u;
-  manager._last_commit_context = std::make_shared<CommitContext>(0u);
+  manager._last_commit_id = 1u;
+  manager._last_commit_context = std::make_shared<CommitContext>(1u);
```
Comment (mjendruk) [inline src/lib/concurrency/transaction_manager.cpp]: Should we replace these numbers with constants? `FIRST_COMMIT_ID`, `INVALID_COMMIT_ID`? Everything else looks good.

Comment (mrks) [reply src/lib/concurrency/transaction_manager.cpp]: Yes, we should. Done.

File: src/lib/concurrency/transaction_manager.hpp [hunk 2]
```diff
@@ -82,7 +82,13 @@ class TransactionManager : private Noncopyable {
 
  private:
   std::atomic<TransactionID> _next_transaction_id;
+  // TransactionID = 0 means "not set" in the MVCC columns
+  static constexpr auto initial_transaction_id = TransactionID{1};
+
   std::atomic<CommitID> _last_commit_id;
+  // We use commit_id=0 for rows that were inserted and then rolled back. Also, this can be used for rows that have
+  // been there "from the beginning of time".
+  static constexpr auto initial_commit_id = CommitID{1};
```
Comment (mjendruk) [inline src/lib/concurrency/transaction_manager.hpp]: Shouldn’t these be in `types.hpp` and all caps like the other constants?

Comment (mrks) [reply src/lib/concurrency/transaction_manager.hpp]: I wouldn't say so. Those are internal to the tx manager and should not be exposed.

File: src/lib/operators/insert.cpp [hunk 3]
```diff
@@ -219,6 +219,12 @@ void Insert::_on_commit_records(const CommitID cid) {
 void Insert::_on_rollback_records() {
   for (auto row_id : _inserted_rows) {
     auto& chunk = _target_table->get_chunk(row_id.chunk_id);
+    // We set the begin and end cids to 0 (effectively making it invisible for everyone) so that the ChunkCompression
+    // does not think that this row is still incomplete. We need to make sure that the end is written before the begin.
+    chunk.mvcc_columns()->end_cids[row_id.chunk_offset] = 0u;
+    std::atomic_thread_fence(std::memory_order_release);
+    chunk.mvcc_columns()->begin_cids[row_id.chunk_offset] = 0u;
```
[the remark this answers is not in the document]
Comment (mrks) [reply src/lib/operators/insert.cpp]: I wouldn't say that we need one. 0 is not a magic number in the way that 1 is.

File: src/lib/concurrency/transaction_manager.hpp (hunk 2 above)
Comment (mjendruk) [reply src/lib/concurrency/transaction_manager.hpp]: okay, and all caps?

File: src/lib/operators/insert.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (mjendruk) [reply src/lib/operators/insert.cpp]: ok

File: src/lib/concurrency/transaction_manager.hpp (hunk 2 above)
Comment (mrks) [reply src/lib/concurrency/transaction_manager.hpp]: done

**Pull request merged**